### DISCUSSION TITLE
OAuth2 Apps

### DISCUSSION THREAD

**Discussion by martmull** [twentyhq/twenty#4116] [Ideas]
Today, external services can connect to Twenty using Api Keys. But we want to go further and connect using OAuth2

The goal is to let any user to connect an external service (like Zapier) using its Twenty account with OAuth2

- follow this doc to implement what is necessary using the Zapier example -> https://platform.zapier.com/build/oauth
- generalize for any external app

## How to?

![Image](https://github.com/twentyhq/twenty/assets/29927851/ac0da8c4-c7ce-47b7-b93a-48d4577c5bbe)
> https://medium.com/google-cloud/understanding-oauth2-and-building-a-basic-authorization-server-of-your-own-a-beginners-guide-cf7451a16f66

**Comment by StephanieJoly4** [comment]
@Bonapara , sharing that this is something a customer would need in the coming months.

Context is that they want their users to connect to their app. Once in their app, they would like them to connect to Twenty via Oauth, to be able to surface data from Twenty that they use more as a backend for a specific type of data.

Happy to discuss further live to see if / how we can include it in the roadmap.

**Comment by BOHEUS** [comment]
Planned for Q2 https://github.com/twentyhq/core-team-issues/issues/16

**Reply by Bonapara** [reply]
might be later this year
